src/engine/CollisionObjectHelpers.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Godot;
+
+/// <summary>
+///   Common helper operations for CollisionObjects
+/// </summary>
+public static class CollisionObjectHelpers
+{
+    /// <summary>
+    ///  Creates and returns a new shape owner for a shape and the given entity.
+    ///  Applies a given transform to the new shapeOwner.
+    /// </summary>
+    /// <returns>Returns the id of the newly created shapeOwner</returns>
+    public static uint CreateShapeOwnerWithTransform(this CollisionObject entity, Transform transform, Shape shape)
+    {
+        var newShapeOwnerId = entity.CreateShapeOwner(shape);
+        entity.ShapeOwnerAddShape(newShapeOwnerId, shape);
+        entity.ShapeOwnerSetTransform(newShapeOwnerId, transform);
+        return newShapeOwnerId;
+    }
+
+    /// <summary>
+    ///  Creates a new shapeOwner for the first shape in the oldShapeOwner of the oldParent
+    ///  and applies a transform to it.
+    ///  Doesn't destroy the oldDhapeOwner.
+    ///  Returns the new ShapeOwnerId.
+    /// </summary>
```

Comment:
This also needs separation as in https://github.com/Revolutionary-Games/Thrive/pull/2700#discussion_r753945252.